pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,608 @@
+	if err != nil {
+		return nil, fmt.Errorf("init StateRoot service: %w", err)
+	}
+
+	netServer.AddExtensibleService(stateRootService, stateroot.Category, stateRootService.OnPayload)
+
+	// do we need oracle service?
+
+	// var oracleService *oracle.Oracle
+	//
+	// if cfg.ApplicationConfiguration.Oracle.Enabled {
+	// 	var c oracle.Config
+	// 	c.Log = log
+	// 	c.Network = cfg.ProtocolConfiguration.Magic
+	// 	c.MainCfg = cfg.ApplicationConfiguration.Oracle
+	// 	c.Chain = artifacts.blockchain
+	// 	c.OnTransaction = artifacts.netServer.RelayTxn
+	//
+	// 	oracleService, err = oracle.NewOracle(c)
+	// 	if err != nil {
+	// 		return fmt.Errorf("init Oracle module: %w", err)
+	// 	}
+	//
+	// 	artifacts.blockchain.SetOracle(oracleService)
+	// 	artifacts.netServer.AddService(oracleService)
+	// }
+
+	var consensusService consensus.Service
+
+	if cfg.ApplicationConfiguration.Consensus.Enabled {
+		var c consensus.Config
+		c.Logger = log
+		c.Broadcast = netServer.BroadcastExtensible
+		c.Chain = blockChain
+		c.ProtocolConfiguration = blockChain.GetConfig().ProtocolConfiguration
+		c.RequestTx = netServer.RequestTx
+		c.StopTxFlow = netServer.StopTxFlow
+		c.Wallet = cfg.ApplicationConfiguration.Consensus.UnlockWallet
+		c.TimePerBlock = cfgServer.TimePerBlock
+
+		consensusService, err = consensus.NewService(c)
+		if err != nil {
+			return nil, fmt.Errorf("init Consensus module: %w", err)
+		}
+
+		netServer.AddConsensusService(consensusService, consensusService.OnPayload, consensusService.OnTransaction)
+	}
+
+	var notaryService *notarysvc.Notary
+
+	if cfg.ApplicationConfiguration.P2PNotary.Enabled {
+		if !blockChain.P2PSigExtensionsEnabled() {
+			return nil, errors.New("P2PSigExtensions are disabled, but Notary service is enabled")
+		}
+
+		var c notarysvc.Config
+		c.MainCfg = cfg.ApplicationConfiguration.P2PNotary
+		c.Chain = blockChain
+		c.Log = log
+
+		notaryService, err = notarysvc.NewNotary(c, netServer.Net, netServer.GetNotaryPool(), func(tx *transaction.Transaction) error {
+			err := netServer.RelayTxn(tx)
+			if err != nil && !errors.Is(err, core.ErrAlreadyExists) {
+				return fmt.Errorf("relay completed notary transaction %s: %w", tx.Hash().StringLE(), err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("init Notary service: %w", err)
+		}
+
+		netServer.AddService(notaryService)
+		blockChain.SetNotary(notaryService)
+	}
+
+	// "make" channel rw to satisfy Start method
+	chErrRw := make(chan error)
+
+	go func() {
+		err, ok := <-chErrRw
+		if ok {
+			chErr <- err
+		}
+	}()
+
+	rpcServer := rpcsrv.New(blockChain, cfg.ApplicationConfiguration.RPC, netServer, nil, log, chErrRw)
+
+	netServer.AddService(&rpcServer) // is it a cyclic dependency?
+
+	res := &Blockchain{
+		base:      blockChain,
+		netServer: netServer,
+		nep17GAS:  nep17.New(_actor, gas.Hash),
+		actor:     _actor,
+		rpcActor:  rpcActor,
+		chErr:     chErrRw,
+	}
+
+	if !cfg.ApplicationConfiguration.RPC.StartWhenSynchronized {
```

Comment:
IIRC NeoFS instances usually use `StartWhenSynchronized` as it allows to avoid replaying old events. For us it means that we can have this set the way we want to in the node.

Reply:
Made behavior with static `true`.